.github/workflows/build_all.yml

```diff
@@ -176,50 +162,42 @@ jobs:
-            echo "$HTTP_RESPONSE"
-            # extract the body and download uri
-            HTTP_BODY=$(echo $HTTP_RESPONSE | sed -e 's/HTTPSTATUS\:.*//g')
-            URI=$(echo $HTTP_BODY | jq -r '.uri')
-            if [[ "$URI" != "null" ]]; then
-              ARTIFACTS=$(echo $ARTIFACTS | jq --arg uri $URI '.packages += [$uri]')
-            else
-              PUBLISH_ERROR="true"
-            fi
-            # extract and check status
-            HTTP_STATUS=$(echo $HTTP_RESPONSE | tr -d '\n' | sed -e 's/.*HTTPSTATUS://')
-            if [[ "$HTTP_STATUS" != "2"* ]]; then
-              PUBLISH_ERROR="true"
-            fi
-            # extract magma debian package version
-            match="magma_[0-9]+\.[0-9]+\.[0-9]+-[0-9]+-[a-z0-9]+_[a-z0-9]+.deb"
-            if [[ $i =~ $match ]]; then
-              magma_package=${i#magma_}
-              magma_package=${magma_package%_[a-z0-9]*.deb}
-              magma_package='magma='${magma_package}
-              echo "magma_package=${magma_package}" >> $GITHUB_OUTPUT
-            fi
+          # extract magma debian package version
+          version_pattern="magma_([0-9]+\.[0-9]+\.[0-9]+-[0-9]+-[a-z0-9]+)_amd64.deb"
+          magma_version=""
+          for i in packages/*.deb; do
+              if [[ $i =~ $version_pattern ]]; then
+                  magma_version=${BASH_REMATCH[1]}
+              fi
           done
+          if [[ -z "$magma_version" ]]; then
+              echo "No file found with a matching version pattern \"${version_pattern}\". Files in folder:"
+              ls -la packages
+              exit 1
+          else
+              echo "Exporting magma version \"${magma_version}\""
+              echo "magma_package=magma=${magma_version}" >> $GITHUB_OUTPUT
+          fi
 
-          #Uploading it to the Linux foundation Registry
-          jf rt upload \
+          # Uploading packages to the registry
+          RESPONSE=$(jf rt upload \
             --recursive=false \
             --detailed-summary \
             --url https://linuxfoundation.jfrog.io/artifactory/ \
             --user ${{ secrets.LF_JFROG_USERNAME }} \
             --password ${{ secrets.LF_JFROG_PASSWORD }} \
             --target-props="${DEBIAN_META_INFO}" \
-            "(*).deb" magma-packages-test/pool/focal-ci/{1}.deb
-          # set output
-          if [[ "$PUBLISH_ERROR" != "true" ]]; then
-            ARTIFACTS=$(echo $ARTIFACTS | jq '. += {"valid":true}')
-          fi
+            "(*).deb" magma-packages-test/pool/focal-ci/{1}.deb)
+
+          # Mapping response to output (for firebase)
+          echo "Response:"
+          echo $RESPONSE | jq
+          echo "Output (for firebase):"
+          ARTIFACTS=$(echo $RESPONSE | jq '{"valid": (.status=="success"), "packages": [.files[] | .target]}')
```

Comment:
:+1:

Reply:
kudos to @jheidbrink - my initial approach to build the output was more procedural-ly